distributed/client.py

```diff
@@ -2571,6 +2571,7 @@ def __anext__(self):
             raise StopAsyncIteration  # flake8: noqa
         while self.queue.empty():
             yield self.event.wait()
+        sleep(0.0000000001)
```

Comment:
@pitrou any thoughts on why this might be necessary on Windows but not Linux?

Adding a sleep of any positive number seemed to work.  Adding a sleep of 0 failed (thing hang).  Adding a print works.  Adding `1 + 1` hangs.

All of these fail on my local VM, so I don't think it's just an issue of CI services being slow enough to catch race conditions.